Userland/Libraries/LibAudio/MP3Loader.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2021, Arne Elster <arne@elster.li>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "MP3Loader.h"
+#include "MP3HuffmanTables.h"
+#include "MP3Tables.h"
+#include <LibCore/File.h>
+#include <LibCore/FileStream.h>
+
+namespace Audio {
+
+MP3LoaderPlugin::MP3LoaderPlugin(StringView path)
+    : m_file(Core::File::construct(path))
+{
+    if (!m_file->open(Core::OpenMode::ReadOnly)) {
+        m_error = LoaderError { String::formatted("Can't open file: {}", m_file->error_string()) };
+        return;
+    }
+
+    off_t file_size = 0;
+    if (!m_file->seek(0, Core::SeekMode::FromEndPosition, &file_size)) {
+        m_error = LoaderError { String::formatted("Could not seek in file.") };
```

Comment:
Also, don't use String::formatted if you're not formatting anything.

Reply:
Same here. Did I break something? 😐